src/kakarot/instructions/system_operations.cairo

```diff
@@ -152,37 +137,182 @@ namespace SystemOperations {
         range_check_ptr,
         bitwise_ptr: BitwiseBuiltin*,
     }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
+        let (ctx, call_args) = CallHelper.prepare_args(ctx=ctx, with_value=1);
+
+        // TODO: use gas_limit when init_at_address is updated
+        let sub_ctx = ExecutionContext.init_at_address(
+            address=call_args.address,
+            calldata_len=call_args.args_size,
+            calldata=call_args.calldata,
+            value=call_args.value,
+            parent_context=ctx,
+            return_data_len=call_args.ret_size,
+            return_data=call_args.return_data,
+        );
+
+        return sub_ctx;
+    }
+
+    // @notice STATICCALL operation.
+    // @dev
+    // @custom:since Homestead
+    // @custom:group System Operations
+    // @custom:gas 0 + dynamic gas
+    // @custom:stack_consumed_elements 6
+    // @custom:stack_produced_elements 1
+    // @return The pointer to the sub context.
+    func exec_staticcall{
```

Comment:
How do we achieve staticcall spec?

> This instructions is equivalent to [CALL](https://www.evm.codes/#F1), except that it does not allow any state modifying instructions or sending ETH in the sub [context](https://www.evm.codes/about). The disallowed instructions are [CREATE](https://www.evm.codes/#F0), [CREATE2](https://www.evm.codes/#F5), [LOG0](https://www.evm.codes/#A0), [LOG1](https://www.evm.codes/#A1), [LOG2](https://www.evm.codes/#A2), [LOG3](https://www.evm.codes/#A3), [LOG4](https://www.evm.codes/#A4), [SSTORE](https://www.evm.codes/#55), [SELFDESTRUCT](https://www.evm.codes/#FF) and [CALL](https://www.evm.codes/#F1) if the [value](https://www.evm.codes/#34) sent is not 0.

I get that we force with_value = 0 to ensure ETH sent with value is 0, but what about the rest? Is this for future implems / diffs?

Reply:
good point. I was too focus single CALL/STATICCALL usage and had in mind that value=0 would break any attempt to something that cost, which is wrong.

A possible implementation that comes to me would be to add a flag "static" in the ExecutionContext and update these opcodes to check if the ExecutionContext is allowed to call them. What do you think?

Reply:
If you want we can split these into two PRs. Maybe the scope of the PR became too wide?